include/deal.II/grid/grid_tools.h

```diff
@@ -1213,6 +1213,11 @@ namespace GridTools
     template <int dim, int spacedim>
     struct DistributedComputePointLocationsInternal
     {
+      /**
+       * Number of searched point locations.
+       */
+      unsigned int n_searched_points;
```

Comment:
I guess we should add an assert that the value is large enough? The local indices in `recv_components` should be all smaller than `n_searched_points`, shouldn't it?

Reply:
At which state should the Assert be placed? I can think about two places
* During sorting of ```recv_``` and ```send_components```
* During the call of ```RPE().reinit()```
 
Also, I forgot to assign an initial value. (TODO for myself)

Reply:
> At which state should the Assert be placed?

I think at the first place it is used.